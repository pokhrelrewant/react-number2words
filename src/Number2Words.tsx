import * as React from 'react';
import { ConverterUtil } from './converterUtil';

interface Props {
  value: number | string;
}

interface State {
  words: string;
}

class Number2Words extends React.Component<Props, State> {

  componentWillMount() {
    const util = new ConverterUtil(this.props.value);
    this.setState({words: util.getWords()});
  }

  render() {
    const { words } = this.state;
    return (
      <span className="number2words__container">{words}</span>
    );
  }
}

export default Number2Words;
