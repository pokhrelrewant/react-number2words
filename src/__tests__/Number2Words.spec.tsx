import * as React from 'react';
import { shallow } from 'enzyme';
import Number2Words from '../Number2Words';

test('The text should be three', () => {
  const component = shallow(<Number2Words value={3}/>);

  expect(component.text()).toMatch('three');

  expect(component).toMatchSnapshot();
});

test('The text should be eight hundred ninety four', () => {
  const component = shallow(<Number2Words value={894} />);

  expect(component.text()).toMatch('eight hundred ninety four');

  expect(component).toMatchSnapshot();
});

const word = 'fourteen billion six hundred million seven hundred twenty eight thousand one hundred ninety five';
test(`The text should be ${word}`, () => {
  const component = shallow(<Number2Words value={14600728195} />);

  expect(component.text()).toMatch(word);

  expect(component).toMatchSnapshot();
});
