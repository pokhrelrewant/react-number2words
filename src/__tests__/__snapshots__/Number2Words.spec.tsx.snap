// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Number2Words shows the current value of the component 1`] = `
<span
  className="number2words__container"
>
  three
</span>
`;

exports[`The text should be eight hundred ninety four 1`] = `
<span
  className="number2words__container"
>
  eight hundred ninety four
</span>
`;

exports[`The text should be fourteen billion six hundred million seven hundred twenty eight thousand one hundred ninety five 1`] = `
<span
  className="number2words__container"
>
  fourteen billion six hundred million seven hundred twenty eight thousand one hundred ninety five
</span>
`;

exports[`The text should be three 1`] = `
<span
  className="number2words__container"
>
  three
</span>
`;
